pkg/local_workflows/report_analytics_workflow.go

```diff
@@ -81,27 +84,53 @@ func reportAnalyticsEntrypoint(invocationCtx workflow.InvocationContext, inputDa
 			return nil, err
 		}
 
-		callErr := callEndpoint(invocationCtx, input, url)
+		payload := input.GetPayload().([]byte)
+
+		normalizeInput(payload)
+
+		callErr := callEndpoint(invocationCtx, payload, url, input.GetContentType())
 		if callErr != nil {
 			err := fmt.Errorf("error calling endpoint for input at index %d: %w", i, callErr)
 			return nil, err
 		}
-
 	}
 	logger.Println(reportAnalyticsWorkflowName + " workflow end")
 	return nil, nil
 }
 
-func callEndpoint(invocationCtx workflow.InvocationContext, input workflow.Data, url string) error {
+func normalizeInput(input []byte) []byte {
+	var scanDoneEvent json_schemas.ScanDoneEvent
+	err := json.Unmarshal(input, &scanDoneEvent)
+	if err != nil {
+		return input
+	}
+
+	durationMs, err := strconv.ParseInt(scanDoneEvent.Data.Attributes.DurationMs, 10, 64)
+	if err != nil {
+		return input
+	}
+
+	if durationMs > 1000*3600*12 || durationMs < 0 {
+		scanDoneEvent.Data.Attributes.DurationMs = "0"
```

Comment:
Question: wouldn't it make sense to indicate an error case with a distinguishable value instead of "0" like "-1"

Reply:
Isn't that the same? 0 ms is impossible anyways.